--- libgeos.py	2013-09-29 14:35:55.000000000 +0100
+++ new.libgeos.py	2013-09-29 14:37:47.000000000 +0100
@@ -110,7 +110,7 @@
     is a release candidate (and what number release candidate), and the C API
     version.
     """
-    ver = geos_version()
+    ver = geos_version().split(' ')[0]
     m = version_regex.match(ver)
     if not m: raise GEOSException('Could not parse version info string "%s"' % ver)
     return dict((key, m.group(key)) for key in ('version', 'release_candidate', 'capi_version', 'major', 'minor', 'subminor'))
